src/uct/ib/base/ib_iface.c

```diff
@@ -255,42 +263,111 @@ int uct_ib_iface_is_reachable(const uct_iface_h tl_iface, const uct_device_addr_
     }
 }
 
-ucs_status_t uct_ib_iface_create_ah(uct_ib_iface_t *iface,
-                                    struct ibv_ah_attr *ah_attr,
-                                    struct ibv_ah **ah_p)
+static int uct_ib_iface_ah_cmp(uct_ib_ah_attr_t *ah_data,
+                               struct ibv_ah_attr *attr,
+                               uct_ib_iface_t *iface)
+{
+    return !memcmp(&ah_data->attr, attr, sizeof(*attr)) &&
+                   (uct_ib_iface_md(iface)->pd == ah_data->pd);
```

Comment:
maybe doing simple compare first would be more optimal?
+alignment

Reply:
in most of cases there is single PD